File: shelfguard_backend/src/models/CustomField.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const CustomField = sequelize.define('CustomField', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  shopId: {
    type: DataTypes.UUID,
    allowNull: false,
    field: 'shop_id',
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  fieldType: {
    type: DataTypes.STRING,
    allowNull: false,
    field: 'field_type',
    validate: {
      isIn: [['text', 'number', 'date', 'boolean', 'select']],
    },
  },
  required: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  selectOptions: {
    type: DataTypes.ARRAY(DataTypes.TEXT),
    field: 'select_options',
  },
  defaultValue: {
    type: DataTypes.TEXT,
    field: 'default_value',
  },
  sortOrder: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 0,
    field: 'sort_order',
  },
}, {
  tableName: 'custom_fields',
  underscored: true,
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
  indexes: [
    {
      unique: true,
      fields: ['shop_id', 'name'],
    },
  ],
});

module.exports = CustomField;

File: shelfguard_app/lib/features/auth/domain/entities/custom_field.dart
import 'package:equatable/equatable.dart';

class CustomField extends Equatable {
  final String id;
  final String shopId;
  final String name;
  final String fieldType; // text, number, date, boolean, select
  final bool required;
  final List<String>? selectOptions; // For select type
  final String? defaultValue;
  final int sortOrder;
  final DateTime createdAt;
  final DateTime updatedAt;

  const CustomField({
    required this.id,
    required this.shopId,
    required this.name,
    required this.fieldType,
    required this.required,
    this.selectOptions,
    this.defaultValue,
    required this.sortOrder,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        shopId,
        name,
        fieldType,
        required,
        selectOptions,
        defaultValue,
        sortOrder,
        createdAt,
        updatedAt,
      ];
}
